node/subsystem/src/messages.rs

```diff
@@ -156,8 +156,11 @@ pub enum CollatorProtocolMessage {
 	CollateOn(ParaId),
 	/// Provide a collation to distribute to validators.
 	DistributeCollation(CandidateReceipt, PoV),
-	/// Fetch a collation under the given relay-parent for the given ParaId.
-	FetchCollation(Hash, ParaId, oneshot::Sender<(CandidateReceipt, PoV)>),
+	/// As a validator may be collated more than one collator on a given `ParaId`
+	/// there may be more than one collation per relay parent. This message allows
+	/// the user to be updated with all collations as they are harvested from the
+	/// relevant collators.
+	FetchCollations(Hash, ParaId, mpsc::Sender<(CollatorId, CandidateReceipt, PoV)>),
```

Comment:
Since collations are heavy, we should not be actively fetching more than one collation unless we need to. It would be better to expose the collators to the candidate selection subsystem and allow it to make a choice of which collations to fetch.

Reply:
ok, but the guide says this:

> When acting on an advertisement, we issue a WireMessage::RequestCollation. If the request times out, we need to note the collator as being unreliable and reduce its priority relative to other collators. And then make another request - repeat until we get a response or the chain has moved on.

this needs to be changed and what needs to happen is this:

1) We need to let candidate selection subsystem know about advertised collations in an async manner (with a `mpsc`) since these advertisements may arrive at any time
1.5) No collations are actually requested by the protocol subsystem on its own.
2) Selection makes its choice and issues a `FetchCollation` and at this point protocol subsystem makes a request.

?

Reply:
Yes, I agree: let candidate selection know about advertised collations with a `mpsc::Sender` with all the information it needs to fetch a particular collation, but don't actually transfer `(CandidateReceipt, PoV)` without an explicit `Fetch`.

Reply:
Ok, I did that